tools/autograd/templates/VariableType.cpp

```diff
@@ -195,6 +195,13 @@ static VariableFlags compute_flags_tmpl(T tensors) {
 using TensorRef = std::reference_wrapper<const Tensor>;
 using TensorRefList = std::initializer_list<TensorRef>;
 
+// ArrayRef is not covariant, so we need to do some work to apply the
+// implicit constructor
```

Comment:
What's an ArrayRef here? I can't see it. Is TensorList an alias to `ArrayRef<at::Tensor>`?

Reply:
ArrayRef is defined in ATen, and it's a sort of general "reference to array-like thing". TensorList is indeed that alias.

Reply:
Yeah I know what an `ArrayRef` is, but I admit my comment wasn't clear. I meant that the comment is confusing for readers that have little ATen experience, because there's no clear connection to the surrounding code. Change ArrayRef to TensorList

Reply:
But you can only speak of the covariance of a type constructor (ArrayRef) not a type (TensorList). What I'll do is mention the alias.

Reply:
That's fine with me. Just do something to make it clear that the comment actually applies to this code

Reply:
To be honest I didn't remember for sure if `TensorList` was a vector like `variable_list` or if it was an `ArrayRef`